db/model/generator.js

```diff
@@ -191,16 +192,36 @@ module.exports = function generator(seq, dataTypes) {
       }, // beforeDestroy
 
       afterCreate(inst /* , opts*/) {
-        if (inst.createdBy) {
-          return new seq.Promise((resolve, reject) =>
-            inst.addWriter(inst.createdBy)
-            .then(() => resolve(inst))
-            .catch((err) => reject(err))
-          );
-        }
+        return Promise.all([
+          Promise.resolve().then(() => {
+            if (inst.currentCollector && inst.isActive) {
+              const newCollector = inst.currentCollector;
+              return hbUtils.trackGeneratorChanges(inst, null, newCollector);
+            }
+          }),
+          Promise.resolve().then(() => {
+            if (inst.createdBy) {
+              return inst.addWriter(inst.createdBy);
+            }
+          }),
+        ]);
 
         return inst;
       }, // afterCreate
+
+      afterUpdate(inst) {
+        let oldCollector = inst.previous('currentCollector');
```

Comment:
check `inst.changed()` for whether currentCollector or isActive were changed before doing anything here... may be able to short circuit and exit without doing anything

Reply:
If currentCollector wasn't changed we still need to track the changes based on whether isActive changed. (line 216 & 218)
And if isActive wasn't changed we still need to check if it's false, in which case we don't track anything. (line 220)